src/app/services/todos-store.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { shareReplay, map } from 'rxjs/operators';
+
+import Todo from 'src/app/models/todo.model';
+import ITodo from 'src/app/models/itodo';
+import { TodosService } from 'src/app/services/todos.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TodosStoreService {
+  private readonly _todos = new BehaviorSubject<ITodo[]>([]);
+
+  readonly todos$ = this._todos.asObservable();
```

Comment:
We can manually specify the type as `Observable<ITodo[]`, if we don't use the `asObservable()` method, right? Otherwise, the type for `todos$` would be `BehaviorSubject<ITodo[]>`

![image](https://user-images.githubusercontent.com/25910474/80515260-5737fa80-898a-11ea-8afe-535a8284697e.png)

Reply:
yeah, you can specify todo as `Observable<ITodo[]>` since `BehaviorSubject` extends `Observable`